src/wovnio/wovnphp/Headers.php

```diff
@@ -198,16 +198,26 @@ public function computePathLang()
             } else {
                 $request_uri = $this->env['REQUEST_URI'];
             }
-            preg_match($rp, $server_name . $request_uri, $match);
-            if (isset($match['lang'])) {
-                $lang_code = Lang::formatLangCode($match['lang'], $this->store);
-                if (!is_null($lang_code)) {
-                    $this->pathLang = $lang_code;
+
+            $full_url = $server_name . $request_uri;
+            $lang_identifier = null;
+            if ($this->store->settings['url_pattern_name'] == 'custom_domain') {
```

Comment:
Do not put all of this code in Headers. Move it to a custom_domain.php file at least. Please keep in mind how it is done in Wovn.cs. I am not sure if this is correct. https://github.com/WOVNio/Wovn.cs/blob/master/wovn.cs/CustomDomainLanguages/CustomDomainLanguagesModel.cs#L22